src/Medidata.ZipkinTracer.Owin/ZipkinTracerOwinExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Web;
+using Microsoft.Owin;
+
+namespace Medidata.ZipkinTracer.Owin
+{
+    internal static class ZipkinTracerOwinExtensions
+    {
+        public static readonly string HttpContextBaseKey = "System.Web.HttpContextBase";
+
+        public static HttpContextBase ToHttpContext(this IOwinContext owinContext) =>
+            owinContext.Environment.ContainsKey(HttpContextBaseKey) ?
+            owinContext.Environment[HttpContextBaseKey] as HttpContextBase :
+            throw new NotSupportedException("Self hosted and non-System.Web-based scenarios are not supported.");
```

Comment:
Actually the whole property can be simplified like
```cs
public static HttpContextBase ToHttpContext(this IOwinContext owinContext) =>
    owinContext.Get<HttpContextBase>(typeof(HttpContextBase).FullName) ??
    throw new NotSupportedException("Self hosted and non-System.Web-based scenarios are not supported.");
```
`IOwinContext.Get<T>` already does the checking of the key for you.
https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin/OwinContext.cs#L88